flink-tests/src/test/java/org/apache/flink/test/streaming/runtime/SinkV2MetricsITCase.java

```diff
@@ -217,43 +215,41 @@ private void assertSinkMetrics(
         assertThat(subtaskWithMetrics, equalTo(numSplits));
     }
 
-    private void assertSinkCommitterMetrics(
-            JobID jobId, int parallelism, Map<String, Long> expected) {
+    private void assertSinkCommitterMetrics(JobID jobId, Map<String, Long> expected) {
         List<OperatorMetricGroup> groups =
                 reporter.findOperatorMetricGroups(
                         jobId, TEST_SINK_NAME + ": " + DEFAULT_COMMITTER_NAME);
-        assertThat(groups, hasSize(parallelism));
```

Comment:
I originally thought, that when starting a job the TMs/Operators are created, and we start serving records only when the full infrastructure is ready.
Turns out, that the TMs start serving records whenever they are ready, and there is no synchronisation point between them. This caused the flakiness of the test.

I have found that 1 out of 10.000 cases not even the metrics group is created when we arrive at this point.
Since the goal is to check that the values of the metrics are correct, I decided to remove this assertion.

Reply:
Thanks for the clarification.